### PR TITLE
[1LP][RFR] Cached Template Loader

### PR DESCRIPTION
**Pull request by psav**
* Introduces a new --use-template-cache option
* Fails over gracefully if there is no cache and pulls from source

### CODE REVIEW

**Comment by mfalesni** [comment]
@psav you should update requrements.txt too :)

**Comment by psav** [comment]
What with? I didn't install anything...

**Comment by mfalesni** [comment]
@psav ah, I though the pytest-cache needs to be installed, never mind :)

File: fixtures/templateloader.py [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+        provider_templates = trackerbot.provider_templates(trackerbot.api())
+        for provider in list_provider_keys():
+            TEMPLATES[provider] = provider_templates.get(provider, [])
+            config.cache.set(provider, TEMPLATES[provider])
+            count += len(TEMPLATES[provider])
+    else:
+        write_line("Using templates from cache...")
+        provider_templates = None
+        for provider in list_provider_keys():
+            templates = config.cache.get(provider, NoTemplates())
```
Comment (RonnyPfannschmidt) [inline fixtures/templateloader.py]: use None

File: fixtures/templateloader.py [hunk 2]
```diff
@@ -1,18 +1,54 @@
...
+    # which may be too tricky right now.
+
+    count = 0
+
+    if not config.getoption('use_template_cache'):
+        write_line("Loading templates from trackerbot...")
+        provider_templates = trackerbot.provider_templates(trackerbot.api())
+        for provider in list_provider_keys():
+            TEMPLATES[provider] = provider_templates.get(provider, [])
+            config.cache.set(provider, TEMPLATES[provider])
```
Comment (RonnyPfannschmidt) [inline fixtures/templateloader.py]: this will probably raise a exception unless a / is in the provider name

**Pull request merged**